beam/src/components/ItemCount.vue

```diff
@@ -33,8 +33,20 @@ const {
 const isCountComplete = computed(() => count.value === denominator)
 
 const validate = (payload: ClipboardEvent | InputEvent | MouseEvent) => {
-	const newValue = Number((payload.target as HTMLElement).innerHTML) || 0
+	const newValue = Number((payload.target as HTMLElement).innerHTML)
+	if (typeof newValue !== 'number' || isNaN(newValue)) {
+		count.value = 0
+		nextTick(() => {
+			;(payload.target as HTMLElement).innerHTML = '0'
+		})
+		return
+	}
 	count.value = Math.min(newValue, denominator)
+	if (denominator > 0) {
+		count.value = Math.min(newValue, denominator)
+	} else {
+		count.value = newValue
+	}
```

Comment:
Since the `NaN` is coming from the `Math.min` operation, I think this can be simplified to just:

```suggestion
	const newValue = Number((payload.target as HTMLElement).innerHTML) || 0
	count.value = denominator ? Math.min(newValue, denominator) : newValue
```

Reply:
@Alchez I tested your suggestion, but it doesn't handle all edge cases, such as when alphabetical characters are entered.

However, I made some improvements to the validation logic:

```ts
const validate = (payload: ClipboardEvent | InputEvent | MouseEvent) => {
	const newValue = Number((payload.target as HTMLElement).innerHTML)
	if (typeof newValue !== 'number' || isNaN(newValue)) {
		nextTick(() => {
			;(payload.target as HTMLElement).innerHTML = count.value.toString() || '0'
		})
		return
	}
	count.value = denominator > 0 ? Math.min(newValue, denominator) : newValue
}
```

demo:

https://github.com/user-attachments/assets/08276b34-946c-4d60-8d1f-cd45e5fb53a6

What do you think?

Reply:
@lauty95 Hmm, is there a use-case where alphabetical characters should be allowed?

Reply:
I don’t think so — I can't think of any use case.
@agritheory , what do you think? Should we allow alphabetical characters in this Vue component?